src/Arcus.EventGrid.Security/EventGridSubscriptionValidator.cs

```diff
@@ -67,7 +69,7 @@ private async Task<HttpResponseMessage> HandleSubscriptionValidationEventAsync(H
 
             var subscriptionEvent = gridMessage.Events.Single();
 
-            var validationCode = subscriptionEvent.Data?.ValidationCode;
+            var validationCode = subscriptionEvent.GetPayload().ValidationCode;
```

Comment:
Null propogation? Can throw null ref

Reply:
Yes, thanks. I even think we should do both. And verify if we don't return `null` in the method since the base class `EventGridEvent` also don't allow `null` to be set as data.